service/matching/tasklist/task_writer.go

```diff
@@ -140,8 +140,8 @@ func (w *taskWriter) appendTask(taskInfo *persistence.TaskInfo) (*persistence.Cr
 			// it to cassandra, just bail out and fail this request
 			return nil, errShutdown
 		}
-	default: // channel is full, throttle
-		return nil, createServiceBusyError("Too many outstanding appends to the TaskList")
+	case <-ctx.Done(): // channel is full, throttle
```

Comment:
I'm not sure I understand the reason for removing the default here: would it not be desirable to return an error when the append channel is full?